dwave/optimization/src/nodes/manipulation.cpp

```diff
@@ -385,6 +427,10 @@ void PutNode::propagate(State& state) const {
                 }
             }
         }
+
+        if (ptr->num_ambiguities()) {
```

Comment:
Nit but having `num_ambiguities()` and checking it again here feels redundant when you're already checking if `ambiguities_` is empty in `resolve_ambiguities()`.

Reply:
It's because actually constructing the two `views()` actually results in a nontrivial amount of jumping around in memory. I am trying to avoid creating unnecessary objects.

Reply:
Ah, I should have read the call here more closely, makes sense.